### PR TITLE
[Doc] Add deployment documentation about CeleryExecutor

### PR DESCRIPTION
**Pull request by jiangxin369**
## What is the purpose of the change

Add deployment documentation about CeleryExecutor

## Brief change log

Add deployment documentation about CeleryExecutor

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: docs/content/deployment/deploying_scheduler.md [hunk 1]
```diff
@@ -36,42 +36,88 @@ steps to initialize configuration and start the Airflow scheduler.
 
 ### CeleryExecutor
 
-1. Prepare broker and backend. Celery needs a broker and a result backend to distribute and store tasks. For more
-   information about setting up the broker and result backend, refer to the exhaustive
-   [Celery documentation on the topic](https://docs.celeryproject.org/en/latest/getting-started/).
+#### Standalone Deployment
 
-2. Set environment variables
+1. Set up Celery broker and result backend. CeleryExecutor take advantage of the [Celery](https://docs.celeryproject.org/en/stable/index.html) to
```
Comment (SteNicholas) [inline docs/content/deployment/deploying_scheduler.md:41]:
```suggestion
1. Set up Celery broker and result backend. `CeleryExecutor` takes advantage of the [Celery](https://docs.celeryproject.org/en/stable/index.html) to 
```

File: docs/content/deployment/deploying_scheduler.md [hunk 2]
```diff
@@ -36,42 +36,88 @@ steps to initialize configuration and start the Airflow scheduler.
...
 ### CeleryExecutor
 
-1. Prepare broker and backend. Celery needs a broker and a result backend to distribute and store tasks. For more
-   information about setting up the broker and result backend, refer to the exhaustive
-   [Celery documentation on the topic](https://docs.celeryproject.org/en/latest/getting-started/).
+#### Standalone Deployment
 
-2. Set environment variables
+1. Set up Celery broker and result backend. CeleryExecutor take advantage of the [Celery](https://docs.celeryproject.org/en/stable/index.html) to 
+   scale out the number of workers. For this to work, you need to setup a broker and a result backend (RabbitMQ, Redis, ...).
```
Comment (SteNicholas) [inline docs/content/deployment/deploying_scheduler.md:42]: It's recommended to tell the users how to setup a broker and a result backend, which could give a link to users.

Comment (SteNicholas) [inline docs/content/deployment/deploying_scheduler.md:42]:
```suggestion
   scale out the number of workers. For this to work, you need to set up a broker and a result backend (RabbitMQ, Redis, ...). 
```

File: docs/content/deployment/deploying_scheduler.md [hunk 3]
```diff
@@ -36,42 +36,88 @@ steps to initialize configuration and start the Airflow scheduler.
...
-1. Prepare broker and backend. Celery needs a broker and a result backend to distribute and store tasks. For more
-   information about setting up the broker and result backend, refer to the exhaustive
-   [Celery documentation on the topic](https://docs.celeryproject.org/en/latest/getting-started/).
+#### Standalone Deployment
 
-2. Set environment variables
+1. Set up Celery broker and result backend. CeleryExecutor take advantage of the [Celery](https://docs.celeryproject.org/en/stable/index.html) to 
+   scale out the number of workers. For this to work, you need to setup a broker and a result backend (RabbitMQ, Redis, ...). 
+   
+2. Change the Airflow configuration. You need to change your airflow.cfg to point the executor parameter to CeleryExecutor and provide the neccess related Celery settings.
```
Comment (SteNicholas) [inline docs/content/deployment/deploying_scheduler.md:44]:
```suggestion
2. Change the Airflow configuration. You need to change your airflow.cfg to set the `executor` parameter to `CeleryExecutor` and provide the neccess related Celery settings.
```

File: docs/content/deployment/deploying_scheduler.md [hunk 4]
```diff
@@ -36,42 +36,88 @@ steps to initialize configuration and start the Airflow scheduler.
...
 
-3. Prepare the Airflow configuration. To enable CeleryExecutor of the scheduler, you need to run the following command
-   on scheduler server and all celery workers to make sure the configurations among hosts are equivalent. The broker URL
-   and result backend you set up should be passed as well.
+#### Deploy with Docker Compose
 
-   ```bash
-   init-airflow-with-celery-executor.sh $MYSQL_CONN $BROKER_URL $RESULT_BACKEND
-   ```
+Deploying CeleryExecutor is a bit complicated, you can follow this guide to quickly start AIFlow and Scheduler with CeleryExecutor within Docker.
```
Comment (SteNicholas) [inline docs/content/deployment/deploying_scheduler.md:66]: `Deploying CeleryExecutor is a bit complicated` could be removed.

File: docs/content/deployment/deploying_scheduler.md [hunk 5]
```diff
@@ -36,42 +36,88 @@ steps to initialize configuration and start the Airflow scheduler.
...
 
-4. Start Celery cluster.
+Note: the Docker Compose below may be not enough to run production-ready Docker Compose AIFlow installation using it.
```
Comment (SteNicholas) [inline docs/content/deployment/deploying_scheduler.md:68]: Where is the Docker Compose below?

File: docs/content/deployment/deploying_scheduler.md [hunk 6]
```diff
@@ -36,42 +36,88 @@ steps to initialize configuration and start the Airflow scheduler.
...
+docker-compose up -d --scale airflow-worker=2
+```
+
+##### Starting Services
+
+Execute following command to start up all services.
+```shell script
+docker exec ai-flow_aiflow-server_1 bash -c "start-all-aiflow-services.sh"
+```
+Once all servers started, you can visit the [Flower web](http://127.0.0.1:5555/) to view the celery workers.
```
Comment (SteNicholas) [inline docs/content/deployment/deploying_scheduler.md:115]: It's recommended to add a picture to show the celery workers in WebUI.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: docs/content/deployment/deploying_scheduler.md [hunk 7]
```diff
@@ -36,42 +36,88 @@ steps to initialize configuration and start the Airflow scheduler.
...
+Make sure you have Docker and Docker Compose installed on your workstation,
+otherwise please follow these steps to install the tools.
 
-   ```bash
-   start-airflow.sh
-   ```
+1. Install [Docker Community Edition (CE)](https://docs.docker.com/engine/install/) on your workstation. Depending on the OS, you may need to configure your Docker instance to use 4.00 GB of memory for all containers to run properly.
+2. Install [Docker Compose](https://docs.docker.com/compose/install/) on your workstation.
+
+Run following commands to check if the tools are installed successfully.
```
Comment (SteNicholas) [inline docs/content/deployment/deploying_scheduler.md:79]:
```suggestion
Run the following commands to check whether the tools are installed successfully:
```

File: docs/content/deployment/deploying_scheduler.md [hunk 8]
```diff
@@ -36,42 +36,88 @@ steps to initialize configuration and start the Airflow scheduler.
...
+##### Before you begin
 
-5. Start the Scheduler
+Make sure you have Docker and Docker Compose installed on your workstation,
+otherwise please follow these steps to install the tools.
 
-   ```bash
-   start-airflow.sh
-   ```
+1. Install [Docker Community Edition (CE)](https://docs.docker.com/engine/install/) on your workstation. Depending on the OS, you may need to configure your Docker instance to use 4.00 GB of memory for all containers to run properly.
```
Comment (SteNicholas) [inline docs/content/deployment/deploying_scheduler.md:76]:
```suggestion
1. Install [Docker Community Edition (CE)](https://docs.docker.com/engine/install/) on your workstation. Depending on the OS, you may need to configure your Docker instance to use at leaset 4GB of memory for all containers to run properly.
```

File: docs/content/deployment/deploying_scheduler.md [hunk 9]
```diff
@@ -36,42 +36,88 @@ steps to initialize configuration and start the Airflow scheduler.
...
+* Workers - The workers who execute the tasks given by the Scheduler.
+* Redis - The broker that forwards messages from Scheduler to worker.
+* Flower -  The [flower app](https://flower.readthedocs.io/en/latest/) for monitoring the environment. It is available at http://localhost:5555.
+* MySQL - The database. Only MySQL is supported in distributed mode for now.
+* HDFS - The hadoop dfs which is used to store the source code or any resources of your projects.
+
+##### Initializing Environment
+
+This command should be executed in the directory that contains docker-compose.yaml you just downloaded.
+It will help start up database, redis, hdfs and 2 workers and also initialize execution environment of all servers.
```
Comment (SteNicholas) [inline docs/content/deployment/deploying_scheduler.md:104]: It's recommended to tell the users how to check whether the environment has been initialized.

File: docs/content/deployment/deploying_scheduler.md [hunk 10]
```diff
@@ -36,42 +36,88 @@ steps to initialize configuration and start the Airflow scheduler.
...
+* AI Flow servers - AI Flow Server, Notification Server and Scheduler deployed in one container.
+* Workers - The workers who execute the tasks given by the Scheduler.
+* Redis - The broker that forwards messages from Scheduler to worker.
+* Flower -  The [flower app](https://flower.readthedocs.io/en/latest/) for monitoring the environment. It is available at http://localhost:5555.
+* MySQL - The database. Only MySQL is supported in distributed mode for now.
+* HDFS - The hadoop dfs which is used to store the source code or any resources of your projects.
+
+##### Initializing Environment
+
+This command should be executed in the directory that contains docker-compose.yaml you just downloaded.
```
Comment (SteNicholas) [inline docs/content/deployment/deploying_scheduler.md:103]: Should this first explain that the operating environment can be initialized by using the following command, and then mention where to execute this command?

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: docs/content/deployment/deploying_scheduler.md (hunk 10 above)
Comment (jiangxin369) [reply docs/content/deployment/deploying_scheduler.md:103]: Agree

File: docs/content/deployment/deploying_scheduler.md [hunk 11]
```diff
@@ -36,43 +36,98 @@ steps to initialize configuration and start the Airflow scheduler.
...
+Once all servers started, you can visit the [Flower web](http://127.0.0.1:5555/) to view the celery workers.
+
+##### Running Example
+```shell script
+docker exec ai-flow_aiflow-server_1 bash -c "python /opt/aiflow/examples/celery/workflows/simple_workflow/simple_workflow.py"
+```
+
 (configuration)=
 
 ## Configuration
```
Comment (Sxnan) [inline docs/content/deployment/deploying_scheduler.md:133]: This section shows a list of all the configurations we added to Airflow to make it works with AIFlow.
->
This section shows a list of configurations you may need to change to Airflow to make it works with AIFlow.

**Review by Sxnan — APPROVED** [review]

**Pull request merged**